src/include/simeng/Elf.hh

```diff
@@ -12,28 +12,83 @@ const char Format32 = 1;
 const char Format64 = 2;
 }  // namespace ElfBitFormat
 
-struct ElfHeader {
-  uint32_t type;
-  uint64_t offset;
-  uint64_t virtualAddress;
-  uint64_t physicalAddress;
-  uint64_t fileSize;
-  uint64_t memorySize;
+// Elf64_Phdr as described in the elf man page. Only contains SimEng relevant
+// information
+
+// An executable or shared object file's program header table is an array of
+// structures, each describing a segment or other information the system needs
+// to prepare the program for execution.  An object file segment contains one or
+// more sections. Program headers are meaningful only for executable and shared
+// object files.  A file specifies its own program header size with the ELF
```

Comment:
"A file specifies its own program header size with the ELF header's e_phentsize and e_phnum members". Isn't it just the e_phentsize which defines the size whilst e_phnum details the number of headers?

Reply:
This is taken directly from the [man page](https://man7.org/linux/man-pages/man5/elf.5.html). You are correct in what you say. Reading again, I am unsure whether the comment is trying to say the total size of all the headers (e_phentsize * e_phnum) or just the size of one header (e_phentsize). Will update for the latter unless you disagree